server/src/main/java/org/elasticsearch/index/mapper/DocumentMapper.java

```diff
@@ -134,7 +138,15 @@ public void validate(IndexSettings settings, boolean checkLimits) {
         }
 
         if (settings.getIndexSortConfig().hasIndexSort() && mappers().nestedLookup() != NestedLookup.EMPTY) {
-            throw new IllegalArgumentException("cannot have nested fields when index sort is activated");
```

Comment:
I wonder if we need to keep this limitation for existing index? This would mean checking the index version to ensure that this mode is not activated on indices with an index sort and already indexed documents.

Reply:
I see, the problem is that existing indexes do not have the parent field set and a mapping update may introduce a nested object. That's correct, we need to check for the version here.